csrc/host_ir/lowering.cpp

```diff
@@ -95,22 +95,27 @@ void lowerSegment(
         auto wait = IrBuilder::create<hir::Wait>(communication);
         hic.pushBackTopLevelExprs(wait);
       }
-    } break;
+      break;
+    }
     case SchedulerType::ExprEval: {
       // Pseudocode:
       // clang-format off
       // ```
       // clone all expressions and store the copies to a list
+      //
       // if no expressions are stream parallelized:
       //   append the list to the top level
       //   return
-      // for each non-input TensorView:
-      //   if it needs an out-of-loop allocation:
-      //     create an Allocate and append it to the top level
```

Comment:
I was able to remove this extra loop because `Allocate`s can be inserted later right before the hir::ForLoop.